midend/parserUnroll.cpp

```diff
@@ -168,7 +168,9 @@ class ParserStateRewriter : public Transform {
         IR::ArrayIndex *newExpression = expression->clone();
         ExpressionEvaluator ev(refMap, typeMap, valueMap);
         auto *value = ev.evaluate(expression->right, false);
-        if (!value->is<SymbolicInteger>()) return expression;
+        if (!value->is<SymbolicInteger>() || expression->right->is<IR::Constant>()) {
```

Comment:
maybe this case can be pushed a bit up, to avoid cloning the expression and invoking the evaluator?